apps/site/lib/site_web/controllers/schedule/line_controller.ex

```diff
@@ -84,15 +87,19 @@ defmodule SiteWeb.ScheduleController.LineController do
         direction_id: conn.assigns.direction_id,
         route_patterns: conn.assigns.route_patterns,
         shape_map: conn.assigns.shape_map,
-        line_diagram: Enum.map(conn.assigns.all_stops, &update_route_stop_data/1)
+        line_diagram:
+          Enum.map(conn.assigns.all_stops, fn stop ->
+            update_route_stop_data(stop, conn.assigns.alerts, conn.assigns.date)
+          end)
       }
     )
   end
 
-  def update_route_stop_data({data, %RouteStop{} = map}) do
+  def update_route_stop_data({data, %RouteStop{id: stop_id} = map}, alerts, date) do
```

Comment:
We should make this a shared function so that someone else doesn't miss the line api version. 

I'd add update_route_stop_data to LineApi and import it here since line api is more specific and more likely to be long lived. You could instead make it a shared module RouteStopData to hold this function. I would personally hold off making a module since I suspect the structure of the data will change as we flesh out this new line diagram. up to you!